libs/langchain/langchain/document_loaders/pdf.py

```diff
@@ -100,6 +108,32 @@ def _is_valid_url(url: str) -> bool:
         parsed = urlparse(url)
         return bool(parsed.netloc) and bool(parsed.scheme)
 
+    @staticmethod
+    def _is_s3_url(url: str) -> bool:
+        """check if the url is S3"""
+        try:
+            result = urlparse(url)
+            if result.scheme == "s3" and result.netloc:
+                return True
+            return False
+        except ValueError:
+            return False
+
+    @staticmethod
+    def _load_s3_file(file_path: str, temp_pdf: str) -> None:
```

Comment:
Thx. 1, 2, 3 and also 4 are done.
It is inconsistent with the http downloader and has duplication for the temp path code, which I would like to refactore in a separate ticket. We could also keep the file in memory instead of downloading locally.